Add Grocery Stores

mdahlhausen opened the pull request:
Pull request overview
---------------------
This PR adds grocery stores to Comstock.
It also updates ComStock to use OpenStudio version 3.10.0 and openstudio-standards version 0.8.3.
Both are done in this PR, as adding grocery stores requires the version updates, and changes to grocery stores change refrigeration in other ComStock building types.

[Change documentation for grocery stores](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/Change%20Documentation/Changes/v3X%20Grocery%20Stores%20and%20Refrigeration/Grocery%20Stores.docx?d=w1b30c70a5af64964916012341da1f2e5&csf=1&web=1&e=JCIDtF)

[Change documentation for OS/os-stds version update](https://nrel.sharepoint.com/:w:/r/sites/comstock/Shared%20Documents/ComStock%20Filing%20Cabinet/ComStock%20Core/Change%20Documentation/Changes/v36%20OpenStudio%203.10.0%20openstudio%200.8.2%20update/OS%20update.docx?d=wa95a336889184eca8f615e8a5ba20895&csf=1&web=1&e=n1MGEA)

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [x] Documentation
 - [x] Infrastructure (includes apptainer image, buildstock batch, dependencies, continuous integration tests)
 - [x] Sampling
 - [x] Workflow Measures
 - [x] Postprocessing

Author pull request checklist:
 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Updated measure .xml(s)
 - [x] Both `options_lookup.tsv` files updated
 - [x] 10k+ test run
 - [x] Change documentation written
 - [x] ComStock documentation updated
 - [x] Changes reflected in example `.yml` files
 - [x] Changes reflected in `README.md` files
 - [x] Added 'See ComStock License' language to first two lines of each code file
 - [ ] All new and existing tests pass the CI **(N/A - CI is not working)**

### Review Checklist
 - [x] Perform a code review on GitHub
 - [x] All related changes have been implemented: data and method additions, changes, tests
 - [x] Reviewed change documentation
 - [x] Ensured code files contain License reference
 - [x] Results differences are reasonable
 - [ ] CI status: all tests pass **(N/A - CI is not working)**

mdahlhausen committed:
add refrigeration to create typical and options lookup

- add the refrigeration technology level input from a buildstock as an input to the create_typical_building_from_model measure.
- add detail to options_lookup.tsv

mdahlhausen committed:
add grocery space types to prototype space type lookup

mdahlhausen committed:
adjust ext light measure method calls to new os stds

mdahlhausen committed:
update os stds method call to add space loads

space_type_apply_internal_loads no longer adds infiltration

mdahlhausen committed:
Update Gemfile

mdahlhausen committed:
add DEER Gro to NIST infiltration measure

mdahlhausen committed:
update ext lighting to use new stds method

Add typical exterior lighting.
Right now there is only one default option. Most exterior lighting is parking lighting, and ComStock used the same value across most vintages.
Update once there is variation in standards for exterior lighting based on technology level.

mdahlhausen committed:
Update Gemfile

LandanT committed:
Add grocery to building types in CBECS mapping for postprocessing

LandanT committed:
Adding Grocery Stores to Naming Mixin

LandanT committed:
Adding Grocery to addressable segment columns

LandanT committed:
Update comstock.py

LandanT committed:
Update comstock_apportionment.py

rHorsey committed:
Fixes to assist Landan in SuperMarket being returned by sims instead of Grocery.

LandanT committed:
Update Options Lookup TSVs to include refrigeration tsvs

LandanT committed:
Create tsvs-v29.zip

LandanT committed:
Update tsv_sampling.py

LandanT committed:
Adding sampling input files

LandanT committed:
Merge branch 'main' into grocery_stores

LandanT committed:
Update tsvs-v29.zip

LandanT committed:
Update sample_input_20250430-2204_9190.csv

LandanT committed:
options look up tsv update

LandanT committed:
updates

LandanT committed:
update

LandanT committed:
update Option Lookup_TSV

LandanT committed:
updated

LandanT committed:
Update options_lookup.tsv

eringold committed:
update profiling to run locally and get simulation time

LandanT committed:
Update tsvs-v29.zip

mdahlhausen committed:
Update Gemfile

Update to openstudio-standards 0.8.2 release

mdahlhausen committed:
update to OS 3.10.0

mdahlhausen committed:
dockerfile updates for OS 3.10.0

mdahlhausen committed:
Update README.md

LandanT committed:
Create sample_input_20250430-2204_9189.csv

LandanT committed:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

LandanT committed:
Delete sample_input_20250430-2204_9190.csv

mdahlhausen committed:
update sampling initilization

remove problematic region 109,hospital

LandanT committed:
Update Gemfile

LandanT committed:
Update Gemfile

mdahlhausen committed:
Merge branch 'main' into grocery_stores

mdahlhausen committed:
Update .gitignore

ignore non-template copies of postprocessing scripts

LandanT committed:
Fixes sampling bucket file issue

mdahlhausen committed:
Merge branch 'main' into grocery_stores

mdahlhausen committed:
.xml file update

mdahlhausen committed:
Update Gemfile

mdahlhausen committed:
Update comstock_apportionment.py

mdahlhausen committed:
Update comstock.py

- overwrite comstock.py with version from main

mdahlhausen committed:
Update compare_runs.py.template

LandanT committed:
uncomment S3 utilties

LandanT committed:
Update compare_comstock_to_cbecs.py.template

LandanT committed:
Update compare_comstock_to_cbecs.py.template

LandanT committed:
Fix figure reference names for envelope figures

LandanT committed:
Updated Tableau Workbook with Documentation Figures

LandanT committed:
Update tsvs-v29.zip

LandanT committed:
Update tsvs-v29.zip

LandanT committed:
Update building_type_cross_datasets.tex

LandanT committed:
Add tableau workbook for documentation graphics

LandanT committed:
Update distribution_definitions.tex

ChristopherCaradonna commented on postprocessing/comstockpostproc/comstock.py:
```diff
@@ -1983,6 +1988,7 @@ def add_building_type_group(self):
         bldg_type_groups = {
             'FullServiceRestaurant': 'Food Service',
             'QuickServiceRestaurant': 'Food Service',
+            'Grocery': 'Food Service',
```
aren't grocery stores "retail"?

LandanT replied:
So grocery stores are considered Food Sales as it relates to CBECs and NEMS/AEO. Its a subset of retail but more specifically is Food sales which aligns with the definition of Convenience Stores as well

ChristopherCaradonna commented on postprocessing/comstockpostproc/comstock.py:
```diff
@@ -202,6 +202,7 @@ def __init__(self, s3_base_dir, comstock_run_name, comstock_run_version, comstoc
                 self.data = self.downselect_imported_columns(self.data)
                 self.rename_columns_and_convert_units()
                 self.set_column_data_types()
+                self.fix_supermarket_building_type_name()
```
Why is this name replacement required for grocery stores only? It seems like this is being used in postprocessing to change the building type name, but why don't we name it correctly at the start?

mdahlhausen replied:
In standards grocery is referenced as supermarket in the code. If we want to call it grocery stores, we need the conversion from the stds name.

ChristopherCaradonna commented on postprocessing/comstockpostproc/comstock.py:
```diff
@@ -1751,6 +1752,10 @@ def set_column_data_types(self):
         self.data = self.data.with_columns(pl.col(self.YEAR_BUILT).cast(pl.Utf8))
 
+    def fix_supermarket_building_type_name(self):
+        # ComStock grocery stores are noted as SuperMarket
```
Can we expand on this comment? It is unclear why we have to convert the name in postprocessing, and not earlier.

mdahlhausen replied:
In standards grocery is referenced as supermarket in the code. If we want to call it grocery stores, we need the conversion from the stds name.

ChristopherCaradonna commented on postprocessing/comstockpostproc/naming_mixin.py:
```diff
@@ -932,7 +933,9 @@ class NamingMixin():
-        'Warehouse': 'warehouse'
+        'Warehouse': 'warehouse',
+        'Grocery': 'grocery',
+        'SuperMarket': 'grocery'
```
Why do we have "SuperMarket" here?

mdahlhausen replied:
In standards grocery is referenced as supermarket in the code. If we want to call it grocery stores, we need the conversion from the stds name.

ChristopherCaradonna commented on postprocessing/comstockpostproc/resources/CBECS_2018_to_comstock_nems_aeo_building_types.csv:
```diff
@@ -7,7 +7,7 @@ High school,Education,Education,SecondarySchool,Education
 Other classroom education,Education,Education,Other,Education
 Convenience store (w/ or w/out gas station),Food sales,Food sales,Other,Food Sales
-Grocery store/food market,Food sales,Food sales,Other,Food Sales
+Grocery store/food market,Food sales,Food sales,Grocery,Food Sales
```
Similar to previous comment - isn't grocery retail, not food service?

LandanT replied:
Grocery is food sales in CBECs and NEMs

ChristopherCaradonna commented on resources/Gemfile:
```diff
@@ -9,13 +9,13 @@ gem 'rubocop', github: 'jmarrec/rubocop', ref: '1.50.0-no_json'
+gem 'openstudio-standards', git: 'https://github.com/NREL/openstudio-standards.git', ref: 'd9e1d2fac02768528ca260274b365284baf8c9ce'
 # gem 'openstudio-standards', path: "C:/GitRepos/openstudio-standards" # Use this format when testing openstudio-standards changes locally
-gem 'openstudio-geb', '= 0.5.0', github: 'LBNL-ETA/Openstudio-GEB-gem', ref: 'afd89e017589479207256ea663fa7ea55ca51c9b'
+gem 'openstudio-geb', git: 'https://github.com/LBNL-ETA/Openstudio-GEB-gem.git', ref: '87da5c2678327b73eef54d2364f46225154bb956'
```
What SHA is this? That's not the one that has the geb gem measures we call for comstock. This is the most recent one we are using: https://github.com/LBNL-ETA/Openstudio-GEB-gem/commit/3f68c3e524bde3d8363e6863824b30ae9b7e04ee

mdahlhausen replied:
This is a more recent SHA for the ComStock measures. You can update as needed in a different PR if necessary.

ChristopherCaradonna replied:
The 87da5c SHA is from the 'master' branch, which isn't what we are using.

The 3f68c3e SHA that I pointed to is from the 'comstock_OS3.8' branch, which is what LBL has used to develop the ComStock version of the GEB geb measures.

ChristopherCaradonna commented on resources/options_lookup.tsv:
Grocery stores are not included in the kitchen equipment distributions, which will make them incompatible with the the electric kitchen upgrade scenario.

mdahlhausen replied:
@LandanT

LandanT replied:
We did not have good data/time to find any on kitchen equipment distributions so we dont have those in the model. They are modelled as gas load in OS. In the future we could try and track down information on these distributions

ChristopherCaradonna reviewed (commented):
See comments in github and linked change documentation.

mdahlhausen committed:
add missing prototype space type reference

mdahlhausen committed:
Merge branch 'main' into grocery_stores

mdahlhausen committed:
Merge branch 'main' into grocery_stores

mdahlhausen committed:
Update Gemfile

mdahlhausen committed:
add enduse comparison

Add comstock total energy by end use bar graph by comstock run. This allows us to quickly compare end use differences between runs.

LandanT committed:
Updating Documentation

LandanT committed:
Fixing AHU nightime vent schedule

Found old data that was used to determine nighttime ventilation schedules. So instead of borrowing retail distributions utilized the distributions that were referenced in Chris' report.

LandanT committed:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

mdahlhausen committed:
Update Gemfile

Use openstudio-standards version 0.8.3

mdahlhausen commented:
@LandanT punchlist for this PR:
- [x] add documentation changes
- [x] resolve merge conflicts with sampling/README.md
- [x] respond to inline comments

LandanT committed:
Add refrigeration specific EUL Graphics and text

mdahlhausen committed:
Merge branch 'main' into grocery_stores

LandanT committed:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

LandanT committed:
Add new Comstock cbecs coverage chart and appropriate text updates

LandanT committed:
Update intro text to state all charts, graphs and text might not be up to date

LandanT committed:
Update tsvs-v29.zip

Noticed efficiency bias was not applied to refrigeration technology level distributions

LandanT committed:
Updating Figures

LandanT committed:
adding references for refrigeration

LandanT committed:
Update 4_9_hvac.tex

LandanT committed:
updating figures

LandanT committed:
Update comstock.py

mdahlhausen committed:
add exhaust fan schedules to models

Commit:
update title

Commit:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

LandanT committed:
Update 3_sampling.tex

LandanT committed:
Update 4_9_hvac.tex

LandanT committed:
update tableau workbook

LandanT committed:
update new figure formatting

Commit:
latex error fixes

LandanT committed:
Update refrigeration_tech_distribution.png

LandanT committed:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

LandanT committed:
Update 3_sampling.tex

LandanT committed:
Delete ~comstock_documentation__33300.twbr

LandanT committed:
fix title line spacing

mdahlhausen committed:
set terrain

The terrain default is quite important for adjusting wind speeds, which influences infiltration, and thus heating. Make this a dynamic field in the future.

mdahlhausen committed:
Merge branch 'main' into grocery_stores

mdahlhausen commented:
@LandanT great job on the documentation updates. Comments on the most recent version you shared:
- Figure 2 - CBECS principal building activity building types. "Not in ComStock" stacked bar chart make bottom-justified and match the same scale as All Buildings stacked bar chart
- Figure 11 - national base occupancy schedules. missing grocery
- Figure 15 - building shape by building type. missing grocery (but not necessary). we can probably get rid of this section
- Figure 23 - weighted floor area by energy code walls. not showing up in the document
- Table 17,18 - design outdoor air rates by building type. include grocery
- Table 25 - Fraction of floor area served by HVAC systems with energy recovery. include grocery
- Figure 49 - presence of air-side economizers. include grocery.

Commit:
fix broken figures and references

LandanT committed:
adjust figure font

LandanT committed:
Merge branch 'grocery_stores' of https://github.com/NREL/ComStock into grocery_stores

LandanT committed:
update tableau workbook

LandanT committed:
Update 4_9_hvac.tex
